src/include/duckdb/storage/data_table.hpp

```diff
@@ -248,14 +258,18 @@ class DataTable {
 private:
 	//! Lock for appending entries to the table
 	mutex append_lock;
-	//! The number of rows in the table
-	atomic<idx_t> total_rows;
 	//! The segment trees holding the various row_groups of the table
 	shared_ptr<SegmentTree> row_groups;
 	//! Column statistics
 	vector<shared_ptr<ColumnStatistics>> column_stats;
 	//! The statistics lock
 	mutex stats_lock;
+	//! The number of rows in the table
+	atomic<idx_t> total_rows;
+	//! The number of rows in the table
+	atomic<int64_t> rows_changed;
```

Comment:
It seems like these new variables (`rows_changed` and `prev_end`) are only set, and not actually used for anything. Can we remove them?

Reply:
These are used through the `DataTable::GetRowsChanged()` and `DataTable::GetPrevEnd` to keep track of any deleted tuples in the row group.